### PR TITLE
[DOCS] Refine doc for kyuubi.authentication

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

Add catalog tree for `kyuubi.authentication` options

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate
image

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
cc @bowenliang123

**Review by bowenliang123 — APPROVED** [review]

**Review by cxzl25 — APPROVED** [review]

**Review by turboFei — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -726,19 +726,37 @@ object KyuubiConf {
...
-      " For SASL authentication, KERBEROS and PLAIN auth type are supported at the same time," +
-      " and only the first specified PLAIN auth type is valid.")
+      "The following tree describes the catalog of each option." +
+      "<ul>" +
+      "  <li><code>NOSASL</code></li>" +
+      "  <li>SASL" +
+      "    <ul>" +
+      "      <li>SASL/PLAIN</li>" +
+      "        <ul>" +
+      "          <li><code>NONE</code></li>" +
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:745]: how about keeping the desc?

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:745]: the main purpose of the tree is to make users clear the catalog of each option, and it was explained above, duplication seems too redundant.

Comment (yaooqinn) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:745]: my mistake, I didn't notice that the descriptions are still there

**Pull request closed without merging**